packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -505,93 +592,47 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     }
   }
 
-  bool get _canComputeIntrinsics => crossAxisAlignment != CrossAxisAlignment.baseline;
-
   double _getIntrinsicSize({
     required Axis sizingDirection,
     required double extent, // the extent in the direction that isn't the sizing direction
     required _ChildSizingFunction childSize, // a method to find the size in the sizing direction
   }) {
-    if (!_canComputeIntrinsics) {
-      // Intrinsics cannot be calculated without a full layout for
-      // baseline alignment. Throw an assertion and return 0.0 as documented
-      // on [RenderBox.computeMinIntrinsicWidth].
-      assert(
-        RenderObject.debugCheckingIntrinsics,
-        'Intrinsics are not available for CrossAxisAlignment.baseline.',
-      );
-      return 0.0;
-    }
+    // Translate the extent into BoxConstraints.
+    final BoxConstraints extentConstraints = switch (sizingDirection) {
+      Axis.horizontal => BoxConstraints(maxHeight: extent),
+      Axis.vertical   => BoxConstraints(maxWidth: extent),
+    };
     if (_direction == sizingDirection) {
       // INTRINSIC MAIN SIZE
-      // Intrinsic main size is the smallest size the flex container can take
-      // while maintaining the min/max-content contributions of its flex items.
-      double totalFlex = 0.0;
-      double inflexibleSpace = 0.0;
-      double maxFlexFractionSoFar = 0.0;
-      RenderBox? child = firstChild;
-      while (child != null) {
-        final int flex = _getFlex(child);
-        totalFlex += flex;
-        if (flex > 0) {
-          final double flexFraction = childSize(child, extent) / _getFlex(child);
-          maxFlexFractionSoFar = math.max(maxFlexFractionSoFar, flexFraction);
-        } else {
-          inflexibleSpace += childSize(child, extent);
+      Size computeIntrinsicMainExtentForChild(RenderBox child, BoxConstraints constraints) {
```

Comment:
I'm not sure I got the intrinsics implementation right (here and below), especially for flex children. The last flex child is allowed to take the remaining main extent so if other flex children did not take all of their allocated fractions the last flex child can end up getting a way more generous main axis constraint than it would get based on its flex factor, so "maintaining the min/max-content contributions of its flex items" like the previous implementation does for cross-extent-in-main-extent-out seems ... incorrect?